### ISSUE TITLE
Upholstery Futon link

### ISSUE THREAD

**Issue by SaschaAdler** [crkn-rcdr/sapindale#3]
Once a user has logged into Sapindale, they can access Upholstery's Futon installation by using their auth token. There should be a link on the home page that looks like this:

```
<a href="https://upholstery.canadiana.ca/cookie?token=$TOKEN>Futon</a>
```

where $TOKEN should be the auth token.

I'll work on having upholstery redirect the above URL to Futon when the token is successfully validated.

**Comment by RussellMcOrmond** [comment]
While it might make sense to support a redirect, I don't think the redirect should always be to Futon.  That cookie URL can be used generally to set the cookie prior to any number of requests to CouchDB (Including tools like the demos, or other external tools we may wish to use, which can then make requests without having to independently support our custom authentication).

**Comment by SaschaAdler** [comment]
#4 is for figuring out how to support demos and other tools with our authentication. I've envisioned Couch API requests to upholstery working by setting the Authentication header. Keep in mind that setting the cookie requires using our custom authentication solution, if not AMSA itself.

As for internal scripts and external tools, I'm not sure what the best approach is. Either we continue to keep Couch open on our internal network, or we author JWTs and attach them to every request (which, of course, has the same overhead as the cookie would).

**Comment by RussellMcOrmond** [comment]
Separate from the Futon question, which is resolved:

With the IIIF demo there was a button you could click on CAP which pointed to https://github.com/crkn-rcdr/cihm-cookie-authorizer .  Further requests could be made to Cantaloupe using a IIIF viewer that was unaware of the authentication.

It is the ability to do something similar with Couch requests that we did with Cantaloupe I'm wanting to have available.

The current https://github.com/crkn-rcdr/d10n demos could work against Upholstery by simply creating a "login to CouchDB" button similar to what we had with the CAP IIIF demo, and then having the config.json point to Upholstery rather than "/couchdb".  I think we'll be stuck keeping these tools around for another 6 months, and I kinda like the idea of moving to https and being agnostic about IP addresses earlier than that (Currently the httpd.conf in d10n has " Require ip 172.16.0.0/12 209.87.229.16/29 207.107.70.190 142.150.181.0/24 10.202").

All that said, having a redirect= get parameter that allowed us to go to other than the Futon default would be handy.

**Comment by SaschaAdler** [comment]
Resolved by #5.
